gcc/rust/expand/rust-macro-builtins.cc

```diff
@@ -287,9 +287,16 @@ MacroBuiltin::concat (Location invoc_locus, AST::MacroInvocData &invoc)
 	}
       else
 	{
-	  rust_error_at (parser.peek_current_token ()->get_locus (),
+	  auto current_token = parser.peek_current_token ();
+	  rust_error_at (current_token->get_locus (),
 			 "argument must be a constant literal");
 	  has_error = true;
+	  if (!parser.skip_token (current_token->get_id ()))
```

Comment:
Oh my, that would be quite the race condition if our parser could not skip the token it's currently looking at! I think a shorter way to do this would be to add an assertion? Because imo we clearly want the compiler to ICE if we ever get into such a situation.
```suggestion
	  rust_assert (parser.skip_token (current_token->get_id ()));
```

If anyone has other input on whether or not we should crash here I'll gladly read it :)

Reply:
> Oh my, that would be quite the race condition if our parser could not skip the token it's currently looking at! I think a shorter way to do this would be to add an assertion? Because imo we clearly want the compiler to ICE if we ever get into such a situation.
> 
> If anyone has other input on whether or not we should crash here I'll gladly read it :)

You are right. In which case, the compiler probably has bigger trouble than I could imagine. Just crashing could reduce the undefined behavior in this situation.